internal/services/aws_reservation_service.go

```diff
@@ -116,14 +113,15 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 		IBClient, ibErr := clients.GetImageBuilderClient(r.Context())
 		logger.Trace().Msg("Creating IB client")
 		if ibErr != nil {
-			renderError(w, r, payloads.NewClientInitializationError(r.Context(), "image builder client", ibErr))
+			renderError(w, r, payloads.NewClientInitializationError(r.Context(), "image builder", ibErr))
 			return
 		}
 
 		// Get AMI
 		ami, ibErr = IBClient.GetAWSAmi(r.Context(), reservation.ImageID)
 		if ibErr != nil {
-			renderError(w, r, payloads.ClientError(r.Context(), "Image Builder", "can't get ami from image builder", ibErr, 500))
+			renderNewErrorFromClientErr(w, r, err)
+			return
```

Comment:
Why are we using renderError and the other time we use renderNewErrorFromClientErr? what is the guideline?